### ISSUE TITLE
Support Multiple Data Sources

### ISSUE THREAD

**Issue by dongjoon-hyun** [cmssnu/dolphin#63]
Hi, let me show you an example usage of multiple data source what I mentioned before. This is just for discussion. :-)

First of all, NN training algorithm needs two kind of data, i.e. prebuilt-model and training data. The training data might be split into two disjoint sets(`test` and `train`) in some algorithms.

Second, there are many options how to provide training data. The most simplest style is label folders containing binary instance files like JPG, MP3, and MP4. The second popular style is using DBMS like LMDB or Level DB. In this issue, let's focus on the first one.

Now, let me describe an specific example of desirable structure for OCR, `mnist`.  (The following path are a local file system or HDFS.)
```
/data/image/mnist/jpg/0
/data/image/mnist/jpg/1
/data/image/mnist/jpg/2
...
/data/image/mnist/jpg/9
/model/mnist/prebuilt-model
/tmp/new-model
```

The `data` folder contains all kinds of data. Here, `0`, ..., `9` are the folder names and will be used a `class` label for the files existing under those folder. `prebuilt-model` is an optional, but it will be used in most of cases during real DNN trainings. `new-model` is a temporary folder for new model snapshots. Please note that many algorithms need to save their model snapshot models every predefined epochs.

`Dolphin` supports multiple stages. So, what about providing their own input dir for each stage?
* model_loading_stage: need to load `/model/mnist/prebuilt-model`
* data_loading_stage: need to load `/data/image/mnist/jpg/*`
* data_splitting_stage: (the path is not needed since it can be hidden)
* training_stage: need to save into `/tmp/new-model`  (We can do this by using Output Service.)

Please leave any comments here.
